crates/env/src/engine/off_chain/impls.rs

```diff
@@ -420,9 +426,28 @@ impl TypedEnvBackend for EnvInstance {
         unimplemented!("off-chain environment does not support contract invocation")
     }
 
+    #[allow(clippy::type_complexity)]
```

Comment:
Maybe we can create an alias to avoid `type_complexity` everywhere=)

Reply:
Nope, the compiler gives a [warning](https://stackoverflow.com/questions/37871267/is-it-possible-to-create-a-type-alias-that-has-trait-bounds-on-a-generic-type-fo) that the type bounds are not checked.